actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/certificate/CertificateActor.java

```diff
@@ -0,0 +1,219 @@
+  public void onReceive(Request request) throws Throwable {
+    Util.initializeContext(request, TelemetryEnvKey.USER);
+    ExecutionContext.setRequestId(request.getRequestId());
+
+    if (CourseActorOperations.ISSUE_CERTIFICATE
+        .getValue()
+        .equalsIgnoreCase(request.getOperation())) {
+      issueCertificate(request);
+    } else {
+      onReceiveUnsupportedOperation(request.getOperation());
+    }
+  }
+
+  private void issueCertificate(Request request) {
+    ProjectLogger.log(
+        "CertificateActor:issueCertificate request=" + request.getRequest(),
+        LoggerEnum.INFO.name());
+    final String batchId = (String) request.getRequest().get(JsonKey.BATCH_ID);
+    final String courseId = (String) request.getRequest().get(JsonKey.COURSE_ID);
+    final String certificateName = (String) request.getRequest().get(CourseJsonKey.CERTIFICATE);
+    final boolean reIssue = isReissue(request.getContext().get(CourseJsonKey.REISSUE));
+    validateCourseBatch(courseId, batchId);
+    Map<String, Object> filters =
+        request.getRequest().containsKey(JsonKey.FILTERS)
+            ? (Map<String, Object>) request.getRequest().get(JsonKey.FILTERS)
+            : new HashMap<>();
+    filters.put(JsonKey.BATCH_ID, batchId);
+    filters.put(JsonKey.ACTIVE, true);
+    SearchDTO searchDTO = new SearchDTO();
+    searchDTO.getAdditionalProperties().put(JsonKey.FILTERS, filters);
+    searchDTO.setLimit(ES_MAX_LIMIT);
+    searchDTO.setFields(Arrays.asList(JsonKey.USER_ID));
+    List<Map<String, Object>> esContents = null;
+    Future<Map<String, Object>> resultF =
+        esService.search(searchDTO, EsType.usercourses.getTypeName());
+    Map<String, Object> result =
+        (Map<String, Object>) ElasticSearchHelper.getResponseFromFuture(resultF);
+    if (MapUtils.isNotEmpty(result)) {
+      esContents = (List<Map<String, Object>>) result.get(JsonKey.CONTENT);
+    }
+    Response response = new Response();
+    Map<String, Object> resultData = new HashMap<>();
+    resultData.put(
+        JsonKey.STATUS, MessageFormat.format(ResponseMessage.SUBMITTED.getValue(), batchId));
+    resultData.put(JsonKey.BATCH_ID, batchId);
+    resultData.put(CourseJsonKey.CERTIFICATE, certificateName);
+    resultData.put(JsonKey.COURSE_ID, courseId);
+    if (CollectionUtils.isEmpty(esContents)) {
+      resultData.put(JsonKey.STATUS, ResponseMessage.NO_USER.getValue());
+    }
+    response.put(JsonKey.RESULT, resultData);
+    sender().tell(response, self());
+    if (CollectionUtils.isNotEmpty(esContents)) {
+      ProjectLogger.log(
+          "CertificateActor:issueCertificate user size=" + esContents.size(),
+          LoggerEnum.INFO.name());
+      esContents
+          .stream()
+          .forEach(
+              userCourse -> {
+                String userId = (String) userCourse.get(JsonKey.USER_ID);
+                try {
+                  pushInstructionEvent(userId, batchId, courseId, certificateName, reIssue);
```

Comment:
Check is there any API to bulk insert events to Kafka.

Reply:
Kafka batching is an internal implementation of kafka producer with property linger.ms, which is configured in below utils PR
https://github.com/project-sunbird/sunbird-utils/pull/587/files